### PR TITLE
feat: Check config value before set into db

### PR DESCRIPTION
**Pull request by yanagiragi**
https://github.com/FlandreDaisuki/dmhy-subscribe/issues/55

* 新增 config 中的 key = `downloader` 時的檢查, 如果設定的 value 不存在則報錯。

* 範例輸出:
```
k45vs@Ragi-PC:/mnt/d/_Repository/GitHub/dmhy-subscribe$ node src/cli/index.mjs config
┌──────────────────────┬───────────────────────────────┐
│         key          │             value             │
├──────────────────────┼───────────────────────────────┤
│      downloader      │            system             │
│ download-destination │                               │
│    aria2-jsonrpc     │ http://localhost:6800/jsonrpc │
│     webhook-url      │                               │
│    webhook-token     │                               │
└──────────────────────┴───────────────────────────────┘
k45vs@Ragi-PC:/mnt/d/_Repository/GitHub/dmhy-subscribe$ node src/cli/index.mjs config downloader aria2c
dmhy:cli:config:setConfig Unsupported downloader
k45vs@Ragi-PC:/mnt/d/_Repository/GitHub/dmhy-subscribe$ node src/cli/index.mjs config
┌──────────────────────┬───────────────────────────────┐
│         key          │             value             │
├──────────────────────┼───────────────────────────────┤
│      downloader      │            system             │
│ download-destination │                               │
│    aria2-jsonrpc     │ http://localhost:6800/jsonrpc │
│     webhook-url      │                               │
│    webhook-token     │                               │
└──────────────────────┴───────────────────────────────┘
k45vs@Ragi-PC:/mnt/d/_Repository/GitHub/dmhy-subscribe$ node src/cli/index.mjs config downloader aria2
Configure successfully
k45vs@Ragi-PC:/mnt/d/_Repository/GitHub/dmhy-subscribe$ node src/cli/index.mjs config
┌──────────────────────┬───────────────────────────────┐
│         key          │             value             │
├──────────────────────┼───────────────────────────────┤
│      downloader      │             aria2             │
│ download-destination │                               │
│    aria2-jsonrpc     │ http://localhost:6800/jsonrpc │
│     webhook-url      │                               │
│    webhook-token     │                               │
└──────────────────────┴───────────────────────────────┘

```

### CODE REVIEW

File: src/cli/commands/config.mjs [hunk 1]
```diff
@@ -28,6 +31,26 @@ export const builder = (yargs) => {
...
+ * @param {string} value
+ */
+const validateConfig = (key, value) => {
+  if (key === 'downloader') {
+    if (!getDownloaders().includes(value)) {
+      return 'DLR_KEY_NOT_FOUND';
+    }
+  }
+  return '';
+};
```
Comment (FlandreDaisuki) [inline src/cli/commands/config.mjs:52]: Validation result should be boolean or explicit expression.
Do not use error key and empty string to implicitly represent validation results.

File: src/cli/commands/config.mjs [hunk 2]
```diff
@@ -28,6 +31,26 @@ export const builder = (yargs) => {
     .example(t('CMD_CONFIG_EXAMPLE3'), t('CMD_CONFIG_EXAMPLE3_DESC'));
 };
 
+const getDownloaders = () => {
+  const thisFileDir = path.dirname(fileURLToPath(import.meta.url));
+  const downloadersDir = path.join(thisFileDir, '..', '..', 'downloaders');
+  const downloaders = fs.readdirSync(downloadersDir).map((fullName) => path.parse(fullName).name);
```
Comment (FlandreDaisuki) [inline src/cli/commands/config.mjs:37]: Do not use `fs` sync API

Comment (yanagiragi) [reply src/cli/commands/config.mjs]: Fixed in [b3c710d](https://github.com/FlandreDaisuki/dmhy-subscribe/pull/58/commits/b3c710d05f2ea783b14948d6f2bd8682a4e1ee1f)

File: src/cli/commands/config.mjs (hunk 1 above)
Comment (yanagiragi) [reply src/cli/commands/config.mjs]: Fixed in [b3c710d](https://github.com/FlandreDaisuki/dmhy-subscribe/pull/58/commits/b3c710d05f2ea783b14948d6f2bd8682a4e1ee1f)

**Comment by FlandreDaisuki** [comment]
resolved #55

**Pull request merged**